bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -470,6 +471,7 @@ void PreProcessor::finalizePreProcessing(NodeIdType clientId) {
   {
     lock_guard<mutex> lock(clientEntry->mutex);
     auto &reqProcessingStatePtr = clientEntry->reqProcessingStatePtr;
+    concord::diagnostics::TimeRecorder scoped_timer(*histograms_.finalizePreProcessing);
```

Comment:
This needs to be added at the beginning of the function to include wait-for-a-lock time, too.

Reply:
Done.